### ISSUE TITLE
False positive for Lint/FormatParameterMismatch when '0' flag and dynamic width are used at the same time

### ISSUE THREAD

**Issue by taichi-ishitani** [bbatsov/rubocop#2540]
Example
```ruby
puts format('%0*x', 2, 1)
```

Rubocop output
```sh
$ cat test.rb
puts format('%0*x', 2, 1)

$ rubocop -V
0.35.1 (using Parser 2.2.3.0, running on ruby 2.2.3 x64-mingw32)

$ rubocop test.rb
Inspecting 1 file
W

Offenses:

test.rb:1:6: W: Number arguments (2) to format mismatches expected fields (0).
puts format('%0*x', 2, 1)
     ^^^^^^

1 file inspected, 1 offense detected
```

I think it is not offense, so Rubocop should not report it.

**Comment by alexdowad** [comment]
Thanks for the report! It will be looked into soon.

**Comment by alexdowad** [comment]
@taichi-ishitani, thank you for reporting this issue. I have pushed a fix to my open PR, which hopefully will be merged soon.

**Comment by taichi-ishitani** [comment]
Hi @alexdowad san,
I have confirmed this issue is resolved with the latest Rubocop.
Thank you for your support!

**Comment by alexdowad** [comment]
どういたしまして。RuboCopを楽しんで使ってください！

**Comment by bbatsov** [comment]
@alexdowad You know Japanese? Impressive! :-)
